Implement service interaction

gyllstromk opened the pull request.

gyllstromk committed:
init

gyllstromk committed:
Improved "passability" of errors to client:

1. Use `message` field instead of `vcapMessage` (restify keys on
        `message`)
2. Include `statusCode` in all `ClientError`s

gyllstromk committed:
Allow client to be passed token instead of user credentials.

alevicki commented on test/services/index.js:
```diff
@@ -0,0 +1,98 @@
+        it('create service succeeds', function (done) {
+            client.services.create(serviceName, 'mongodb', '2.4', 'free',
+                function (err) {
+
```
Unnecessary whitespace

alevicki commented on test/services/index.js:
```diff
@@ -0,0 +1,98 @@
+            });
+        });
+
+        it('get services includes new service', function (done) {
```
You don't actually check that the result includes the new service

alevicki commented on test/services/index.js:
```diff
@@ -0,0 +1,98 @@
+            });
+        });
+
+        it('get starts empty', function (done) {
```
Change name to get returns empty after delete

alevicki commented:
It doesn't seem like nock is the best solution here, but it is also unclear what the best solution would be.  

How feasible is it to require a local micro install of Cloud Foundry?

gyllstromk committed:
Fixes for PR:

1. Test result on GET services after creating service.
2. More accurate title: "get returns empty"

gyllstromk committed:
fix jshint
